blocks/form/checkbox.js

```diff
@@ -0,0 +1,30 @@
+function update(checkboxgroup) {
+  const allUnchecked = checkboxgroup.every((checkbox) => !checkbox.checked);
+  checkboxgroup.forEach((checkbox) => {
+    if (allUnchecked) {
+      checkbox.setAttribute('required', '');
+    } else {
+      checkbox.removeAttribute('required');
+    }
+  });
+}
+
+export default async function decorate(form) {
+  const checkboxes = form.querySelectorAll('input[type=checkbox][required]');
+  const checkboxNameMap = [...checkboxes].reduce((map, checkbox) => {
+    if (map.has(checkbox.name)) {
+      map.get(checkbox.name).push(checkbox);
+    } else {
+      map.set(checkbox.name, [checkbox]);
+    }
+    return map;
+  }, new Map());
+
+  Array.from(checkboxNameMap.values())
+    .filter((checkboxgroup) => checkboxgroup.length > 1)
+    .forEach((checkboxgroup) => {
+      checkboxgroup.forEach((checkbox) => {
+        checkbox.onchange = () => update(checkboxgroup);
+      });
+    });
```

Comment:
Another way could be to identify checkboxes with same name.

```
[...new Set([...checkboxes].map(x.name))].forEach(name => {
   const cbs = document.getElementsByName(name);
   if (cbs.length > 1) 
            cbs.forEach(checkbox => checkbox.addEventListener('change', update())
})
```

Reply:
I will adjust the code with this approach.

Reply:
`getElementsByName` is available only on document. I think the same named checkboxes validation should be done per form level. Moving back to `map` will incorporate above implementation of map/object.